server/app.py

```diff
@@ -96,40 +100,49 @@ def Login():
 with the details of 8 pre-determined works of art so that they can be formatted for gameplay.
  """
 
-
-def fetchArtInformation(artObjectIDs):
-    # Define index, and Cardset dictionary
-    idx = 0
-    cardSet = {}
-    # This for loop goes through all IDs provided in the list of art pieces chosen and pulls the required information about them from the MET's API
-    for i in artObjectIDs:
-        apiResponse = requests.get(
-            "https://collectionapi.metmuseum.org/public/collection/v1/objects/" + str(i)
-        )
-        artDetails = apiResponse.json()
-        # We add another object to the Cardset dictionary for each art piece, containing info on Name, ObjectID, URL and status
-        cardSet[idx] = {
-            "cardId": idx + 1,
-            "ObjectID": str(i),
-            "artName": artDetails["title"],
-            "artURL": artDetails["primaryImage"],
-            "active": False,
-            "status": False,
-        }
-        idx += 1
-    # Returns a nested Json object, with multiple artCards inside
-    return json.dumps(cardSet)
+def fetchArtInformation(i):
+#Numbers from 1 to number of images, identified by the number of processes created
+    idx=current_process()._identity[0]
+#pulls the required information from the MET's API
+    apiResponse = requests.get("https://collectionapi.metmuseum.org/public/collection/v1/objects/" + str(i))
+    artDetails = apiResponse.json()
+#We add another object to the Cardset dictionary for each art piece, containing info on Name, ObjectID, URL and status
+    cardSet={
+    'cardId': idx,
+    'ObjectID': str(i),
+    'artName': artDetails["title"],
+    'artUrl': artDetails["primaryImage"],
+    'active':  False,
+    'status': False
+    }
+#Returns a Json object
+    return (cardSet)
 
 
 @app.route("/api/MetAPI")
 def pullMETAPI():
     print("Met API Method:")
 
-    # These are the ObjectIDs of 8 pieces we selected for this demo.
-    # For future iterations of the game, these objectIDs will need to be selected by the system.
-    artObjectIDs = [16577, 436944, 437879, 436101, 40081, 437329, 436840, 435882]
-    return fetchArtInformation(artObjectIDs)
-
+#These are the ObjectIDs of 8 pieces we selected for this demo.
+#For future iterations of the game, these objectIDs will need to be selected by the system.
+    artObjectIDs = [16577,436944,437879,436101,40081,437329,436840,435882]
+    numPieces = len(artObjectIDs)
+   
+   #Multiprocessing here
+    p=Pool(numPieces)
+    artPieces=p.map(fetchArtInformation,artObjectIDs)
+
+    for x in range(numPieces):
+        cardSet={
+           'cardId': x+numPieces+1,
```

Comment:
The card Id is being set incorrectly... not sure the solution atm

Reply:
Are you sure? I double checked to make sure it would go from 1-16 in our test case

Reply:
So Abi and I talked about this in person, and I will outline the details here:

Since we are using multiprocessing there is no way to increment the card indexes to match (no for-loops). I didn't want to hard code any values because currently all of the API code is able to accommodate variable gameboard sizes/number of pieces. 

Since the value for the CardId is set by multiprocessing, if you refresh the page multiple times it will extend from 1-16 upwards. When I wrote the program I did not determine this to be an issue, however, Abi brought up that the code she wrote for the game board uses the cardIds to set up the page. Players are not meant to refresh the page in gameplay so we have determined this to be an edge-case to revisit. At this time we will proceed with this and determine steps to resolve with the next sprint.

Reply:
@FatimahAreola @CarlaLeal @m2qaderi for visibility:

This is an odd edge case, I think we need to discuss how we want to handle the case when a user "reloads" the page in the game etc. we may need to explore having a different card id index for the multithreading and then forcing the cardId to increment from 1 to board size. for now lets merge and we can discuss how we want to solution in next sprint.